### ISSUE TITLE
Warning generating trace from example.churn

### ISSUE THREAD

**Issue by vschiavoni** [splay-project/splay#53]
ruby 1.8.7 (2013-06-27 patchlevel 374) [i686-darwin13.3.0]
treetop 1.6.3 

How to reproduce:
```bash
cd splay/tools/churn_gen
ruby -rubygems churn_gen.rb -i example.churn -o /tmp/example.trace
WARNING: syntax error in line 1 (at 0s set replacement ratio to 50%): . Ignoring line.
```
The output file is not written.

The input file is this one:
https://github.com/splay-project/splay/blob/master/tools/churn_gen/example.churn

**Comment by vschiavoni** [comment]
This can be reproduced with different versions of treetop (tested with treetop 1.4.15, 1.5.3 and 1.6.3).

**Comment by vschiavoni** [comment]
The following patch seems to solve the problem:
[churn_lang_patch.txt](https://github.com/splay-project/splay/files/38934/churn_lang_patch.txt)

Try to apply and check if it works: in my tests, I could generate the trace from the example.churn input file.

**Comment by vschiavoni** [comment]
To apply the patch:
```bash
git clone https://github.com/splay-project/splay.git
cd splay
patch tools/churn_gen/churn_lang.rb churn_lang_patch.txt
```
Then you should use the churn_gen.rb script in the tools/churn_gen/ directory to generate the desired churn traces.

**Comment by etriviere** [comment]
Why would one apply a patch to a file that is generated? churn_lang.rb is the output of the treetop compiler. It will be erased by re-compiling it.

**Comment by vschiavoni** [comment]
Then, why is the file churn_lang.rb is committed, if it has to be regenerated  ?

**Comment by etriviere** [comment]
Because it allows users to immediately use the generator without having to regenerate. But I agree this can create confusion. OTOH, the code does not look as human-produced code (or I would have a twisted mind).

I am testing it now

**Comment by etriviere** [comment]
You can also note that the first line of the file says:

"# Autogenerated from a Treetop grammar. Edits may be lost.�

I don�t see how we can be clearer. I am adding a BUILD file to the repository just in case.

**Comment by vschiavoni** [comment]
Closed by 4bdae6e0a0174e12d22f3d4b1242cee07e5c7b1d
